recipes/redis-plus-plus/all/conanfile.py

```diff
@@ -11,129 +16,137 @@ class RedisPlusPlusConan(ConanFile):
 
     @property
-    def _build_subfolder(self):
-        return "build_subfolder"
+    def _compilers_minimum_version(self):
+        if Version(self.version) < "1.3.0":
+            return {}
+        return {
+            "Visual Studio": "16",
+            "msvc": "192",
+            "gcc": "8",
+            "clang": "7",
+            "apple-clang": "12",
+        }
 
     def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
+        if Version(self.version) < "1.3.0":
+            del self.options.build_async
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
         self.requires("hiredis/1.0.2")
-        if self.options.build_async:  
-            self.requires("libuv/1.44.1")
+        if self.options.get_safe("build_async"):
+            self.requires("libuv/1.44.2")
 
     def validate(self):
-        if self.settings.compiler.get_safe("cppstd"):
-            if tools.Version(self.version) >= "1.3.0":
-                tools.check_min_cppstd(self, 17)
-            else:
-                tools.check_min_cppstd(self, 11)
-
-        if tools.Version(self.version) >= "1.3.0":
-            minimum_version = self._compiler_required_cpp17.get(str(self.settings.compiler), False)
-            if minimum_version:
-                if tools.Version(self.settings.compiler.version) < minimum_version:
-                    raise ConanInvalidConfiguration("{} requires C++17, which your compiler does not support.".format(self.name))
-            else:
-                self.output.warn("{0} requires C++17. Your compiler is unknown. Assuming it supports C++17.".format(self.name))
-
-        if self.options.with_tls != self.options["hiredis"].with_ssl:
-            raise ConanInvalidConfiguration("with_tls must match hiredis.with_ssl option")
+        if self.info.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._min_cppstd)
+
+        minimum_version = self._compilers_minimum_version.get(str(self.info.settings.compiler), False)
+        if minimum_version and Version(self.info.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support.",
+            )
+
+        if self.info.options.with_tls and not self.dependencies["hiredis"].options.with_ssl:
+            raise ConanInvalidConfiguration(f"{self.name}:with_tls=True requires hiredis:with_ssl=True")
```

Comment:
```suggestion
            raise ConanInvalidConfiguration(f"{self.ref}:with_tls=True requires hiredis:with_ssl=True")
```

Reply:
it's not a valid pattern for command line or profile so I don't recommend this modification

Reply:
In 2.0 you need to pass the version? 🤔 not important